source/macos/AdaptiveCards/AdaptiveCards/Renderers/TextInputRenderer.swift

```diff
@@ -20,17 +20,35 @@ class TextInputRenderer: NSObject, BaseCardElementRendererProtocol {
         if let maxLen = inputBlock.getMaxLength(), Int(truncating: maxLen) > 0 {
             textView.maxLen = Int(truncating: maxLen)
         }
+        let action = inputBlock.getInlineAction()
+        var renderButton: Bool = false
+        switch action {
+        case is ACSOpenUrlAction:
+            renderButton = true
+        case is ACSToggleVisibilityAction:
+            renderButton = true
+        case is ACSSubmitAction:
+            renderButton = true
+        default:
+            renderButton = false
+        }
         
         if inputBlock.getIsMultiline() {
             let multilineView = ACRMultilineInputTextView()
             if let placeholderString = inputBlock.getPlaceholder() {
                 multilineView.setPlaceholder(placeholder: placeholderString)
+                view.placeholder = placeholderString
             }
             if let valueString = inputBlock.getValue(), !valueString.isEmpty {
                 multilineView.setValue(value: valueString, maximumLen: inputBlock.getMaxLength())
+                view.stringValue = multilineView.textView.string
             }
             multilineView.maxLen = inputBlock.getMaxLength() as? Int ?? 0
-            return multilineView
+            view.addArrangedSubview(multilineView)
+            if renderButton {
+                addInlineButton(parentview: view, view: multilineView, element: inputBlock, style: style, with: hostConfig)
```

Comment:
stackView can be returned only if there is need for renderButton else multilineView can be returned.